engine/components-api/src/main/java/com/cloud/network/IpAddressManager.java

```diff
@@ -233,4 +235,52 @@ List<IPAddressVO> listAvailablePublicIps(final long dcId,
 
     public static final String MESSAGE_ASSIGN_IPADDR_EVENT = "Message.AssignIpAddr.Event";
     public static final String MESSAGE_RELEASE_IPADDR_EVENT = "Message.ReleaseIpAddr.Event";
+
+
+    /**
+     * Checks if the given public IP address is not in active quarantine.
+     * It returns `true` if:
+     *  <ul>
+     *   <li>The IP was never in quarantine;</li>
+     *   <li>The IP was in quarantine, but the quarantine expired;</li>
+     *   <li>The IP is still in quarantine; however, the new owner is the same as the previous owner, therefore, the IP can be allocated.</li>
+     * </ul>
+     *
+     * It returns `false` if:
+     * <ul>
+     *   <li>The IP is in active quarantine and the new owner is different from the previous owner.</li>
+     * </ul>
+     *
+     * @param ip used to check if it is in active quarantine.
+     * @param account used to identify the new owner of the public IP.
+     * @return true if the IP can be allocated, and false otherwise.
+     */
+    boolean checkIfPublicIpAddressIsNotInQuarantineAndCanBeAllocated(IpAddress ip, Account account);
```

Comment:
as commented in the implementation:
```suggestion
    boolean canPublicIpAddressBeAllocated(IpAddress ip, Account account);
```